packages/vite/src/node/plugins/css.ts

```diff
@@ -556,16 +563,19 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
           })
           chunk.viteMetadata.importedCss.add(this.getFileName(fileHandle))
         } else if (!config.build.ssr) {
-          // legacy build and inline css
-
-          // __VITE_ASSET__ and __VITE_PUBLIC_ASSET__ urls are processed by
-          // the vite:asset plugin, don't call resolveAssetUrlsInCss here
+          // legacy build, inline css
+          // the entry css(style.css) will be inlined in the html template
+          // so don't need to generate the css insert script
+          if (chunk.isEntry) {
+            return { code }
+          }
```

Comment:
IIUC This branch is for a different bug (https://github.com/vitejs/vite/pull/9507#discussion_r939012550). Is this fixing https://github.com/vitejs/vite/issues/2062#issuecomment-1050349925?

Maybe we should split this diff into another PR.